components/database.js

```diff
@@ -175,6 +175,24 @@ export default class Database {
     }
   }
 
+  static async removePost(id) {
+    let newData = [];
+    try {
+      await this.getItem("data").then((oldData) => {
+        oldData.forEach((post) => {
+          if (post.id != id) {
+            newData.push(post);
+          }
+        });
+      });
+
+      this.setItem(newData, "data");
```

Comment:
Needs await or then error handler for a boolean return statement to make sense. Right now if setitem makes an error it will still return true.